File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>
#include <ctype.h>
#include <string.h>

//declaring functions, described below.
int countl(string word);
int countw(string word);
int counts(string word);
float average(int number1, int number2);

//calculate readability score for text entered by user.
int main(void)
{
    //Prompt user to enter sample text here.
    string text = get_string("Text: ");

    //Use count functions (described below) to determine number of letters, words and sentences.
    int count_letters = countl(text);

    int count_words = countw(text);

    int count_sentences = counts(text);

    //Use average function (described below) to determine averages for readability test (Coleman-Liau index).
    float L = average(count_letters, count_words);

    float S = average(count_sentences, count_words);

    //Use Coleman-Liau index to determine reading level
    float index = 0.0588 * L - 0.296 * S - 15.8;
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(index));
    }
}


//count number of letters in text entered by user.
int countl(string word)
{
    int sum = 0;
    for (int i = 0, n = strlen(word); i < n; i++)
    {
        if ((word[i] >= 'a' && word[i] <= 'z') || (word[i] >= 'A' && word[i] <= 'Z'))
        {
            sum = sum + 1;
        }
    }
    return sum;
}

//count number of words in text entered by user.
int countw(string word)
{
    int sum = 0;
    for (int i = 0, n = strlen(word); i < n; i++)
    {
        if (word[i] == ' ')
        {
            sum = sum + 1;
        }
    }
    sum = sum + 1;
    return sum;
}


//count number of sentences in text entered by user.
int counts(string word)
{
    int sum = 0;
    for (int i = 0, n = strlen(word); i < n; i++)
    {
        if (word[i] == '.' || word[i] == '!' || word[i] == '?')
        {
            sum = sum + 1;
        }
    }
    return sum;
}

//find average number of letters or sentences per 100 words.
float average(int number1, int number2)
{
    float avg;
    avg = (100 * number1) / (float) number2;
    return avg;
}

File: pset6/readability/readability.py
from cs50 import get_string


def main():
    # Prompt user to enter sample text
    text = get_string("Text: ")
    # Employ count functions to determine number of letters, words, and sentences
    countletters = count1(text)

    countwords = countw(text)

    countsentences = counts(text)

    # Use average function to determine averages
    L = average(countletters, countwords)

    S = average(countsentences, countwords)

    # Use Coleman-Liau index to determine reading level
    index = 0.0588 * L - 0.296 * S - 15.8
    if index < 1:
        print("Before Grade 1")
    elif index > 16:
        print("Grade 16+")
    else:
        print(f"Grade {int(round(index))}")


def count1(word):
    summ = 0
    for c in word:
        if (c >= 'a' and c <= 'z') or (c >= 'A' and c <= 'Z'):
            summ += 1

    return summ


def countw(word):
    # Count letters
    summ = 0
    for c in word:
        if c == ' ':
            summ += 1
    summ += 1

    return summ


def counts(word):
    # Count sentences
    summ = 0
    for c in word:
        if (c == '.') or (c == '!') or (c == '?'):
            summ += 1

    return summ


def average(num1, num2):
    # Find average
    avg = ((100 * num1) / num2)
    return avg


main()